Modules/vc-module-inventory/VirtoCommerce.InventoryModule.Data/Services/FulfillmentCenterSearchService.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Caching.Memory;
+using VirtoCommerce.InventoryModule.Core.Model;
+using VirtoCommerce.InventoryModule.Core.Model.Search;
+using VirtoCommerce.InventoryModule.Core.Services;
+using VirtoCommerce.InventoryModule.Data.Cashing;
+using VirtoCommerce.InventoryModule.Data.Repositories;
+using VirtoCommerce.Platform.Core.Caching;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Data.Infrastructure;
+
+namespace VirtoCommerce.InventoryModule.Data.Services
+{
+    public class FulfillmentCenterSearchService : IFulfillmentCenterSearchService
+    {
+        private readonly Func<IInventoryRepository> _repositoryFactory;
+        private readonly IPlatformMemoryCache _platformMemoryCache;
+
+        public FulfillmentCenterSearchService(Func<IInventoryRepository> repositoryFactory, IPlatformMemoryCache platformMemoryCache)
+        {
+            _repositoryFactory = repositoryFactory;
+            _platformMemoryCache = platformMemoryCache;
+        }
+
+        public async Task<GenericSearchResult<FulfillmentCenter>> SearchCentersAsync(FulfillmentCenterSearchCriteria criteria)
+        {
+            var cacheKey = CacheKey.With(GetType(), "SearchCentersAsync", criteria.GetHashCode().ToString());
```

Comment:
Need to replace everywhere criteria.GetHashCode()  or other ValueObject in a cache key constuction expressions to criteria.ToString();